features/mbedtls/targets/hash_wrappers.c

```diff
@@ -0,0 +1,356 @@
+#include "mbedtls/md2.h"
```

Comment:
If we are importing a Mbed TLS 2.7.1 and adapting mbed-os at the same time. As in this PR. We should not need these wrappers. Please clarify their requirement.

Reply:
When the MBEDTLS_*_ALT symbols are defined almost entire translation unit for the given hash is compiled out, therefore whenever _ALT implementation is provided, it must come along with the wrappers. In the proposed approach the wrappers are common for all targets in order to avoid doing this for each target separately.

Reply:
Lets take an example ```mbedtls_md2_starts_ret``` (new) and ```mbedtls_md2_starts``` (old) are both flavors of the same API. New ```mbedtls_md2_starts_ret``` is called from ```md2.c```. 
Where is ```mbedtls_md2_starts``` called from?

Reply:
As long as the deprecated API has not been removed I believe that any Mbed OS application may call it. It has been pointed out in other PR: https://github.com/ARMmbed/mbed-os/pull/5973#issuecomment-367373634

Reply:
> As long as the deprecated API has not been removed I believe that any Mbed OS application may call it. It has been pointed out in other PR: [#5973 (comment)](https://github.com/ARMmbed/mbed-os/pull/5973#issuecomment-367373634)
I believe @LiyouZhou test code is to demonstrate the missing symbols issue in that PR. @LiyouZhou do you have a requirement to use deprecated API. It is not recommended to promote deprecated API.

Reply:
So in my case the problem is when customer do their own implementation they only port the new API but not the old. That mean all old API will be broken on that particular platform. I think it is a requirement the old API not be broken all of a sudden.

Reply:
Hi @LiyouZhou this PR updates the API as well as the implementation. Hence there are no old API calls from within the library. 

Question to is about the applications. Do you have an application that you think will break with new API. We recommend updating the application when you decide to update mbed-os version in it.